pkg/specgen/generate/container_create.go

```diff
@@ -119,6 +120,18 @@ func MakeContainer(ctx context.Context, rt *libpod.Runtime, s *specgen.SpecGener
 	}
 
 	newImage, resolvedImageName, imageData, err := getImageFromSpec(ctx, rt, s)
+
+	conf, err := rt.GetConfig()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if imageData != nil {
+		crunVariant := config.ImagePlatformToRuntimes(imageData.Os, imageData.Architecture, conf.Engine.PlatformToOCIRuntime)
+		if len(crunVariant) > 0 {
+			rt.UpdateDefaultOCIRuntime(imageData.Os+"/"+imageData.Architecture)
```

Comment:
@mheon now runtimes are already pre-populated but I want to switch default `runtime` automagically on the basis of its name. Is is safe to update default runtime ? If not could you suggest alternative ?

Reply:
Nvm this is resolved.